src/Tests/dotnet-workload-list.Tests/GivenAnMsiInstallation.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using Microsoft.DotNet.Workloads.Workload.Install.InstallRecord;
+using Microsoft.NET.TestFramework;
+using Microsoft.Win32;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.Workload.List.Tests
+{
+    [Collection("MsiWorkloadRecords")]
+    public class GivenAnMsiInstallation
+    {
+        // Override HKLM to HKCU so we can run tests without needing elevation
+        private static MsiWorkloadInstallationRecordManager RecordManager = new MsiWorkloadInstallationRecordManager(
+            Registry.CurrentUser,
+            @"SOFTWARE\Microsoft\dotnet-test\InstalledWorkloads\Standalone");
+
+        [WindowsOnlyFact]
+        public void ItCanReadWorkloadRecords()
+        {
+            try
+            {
+                CreateWorkloadRecord("6.0.100", "workload.A");
+                CreateWorkloadRecord("6.0.100", "workload.B");
+                CreateWorkloadRecord("6.0.200", "workload.C");
+
+                var records = RecordManager.GetInstalledWorkloads(new SdkFeatureBand("6.0.200"));
+
+                Assert.Contains(new WorkloadId("workload.C"), records);
+            }
+            finally
+            {
+                Registry.CurrentUser.DeleteSubKeyTree(@"SOFTWARE\Microsoft\dotnet-test");
```

Comment:
This is to clean up the registry, right? Isn't there a built in way to add a dispose method to run after tests?

Reply:
you mean in the SDK test framework? I don't know, I'll look around

Reply:
@wli3 might know?

Reply:
Looks like I can just implement IDisposable based on what I read